### PR TITLE
Recipes responder skeleton

### PR DESCRIPTION
**Pull request by fredex42**
## What does this change?

https://trello.com/c/xuSwPkfG

- importing the content-api-models npm package
- set up a new lambda function
- use this code (in the lambda function package) to start off unmarshalling and parsing the content: https://github.com/guardian/pubflow-two/blob/main/packages/lambda/crier-responder/src/main.ts
- (also in the lambda function package) supply stub definitions for handleContentUpdate, handleContentUpdateRetrievable, handleDeletedContent

## How to test

There's a basic test of the decoder in-line.  It's kinda hard to directly test because the functions that get called are all stubs which throw exceptions; however, you could deploy it to CODE and watch the lambda run.  Check the logs to see that the expected exceptions get called, then either disable the kinesis trigger on the lambda or delete it by rolling back

## How can we measure success?

In a position to start building out the backend functionality

## Have we considered potential risks?

n/a

### CODE REVIEW

**Review by JustinPinner — APPROVED** [review]
LGTM. We appear to be testing that we can listen to the Kinesis flow and have yet to handle the actual updates/deletes etc so 👍

File: cdk/lib/recipes-backend.ts [hunk 1]
```diff
@@ -42,23 +42,29 @@ export class RecipesBackend extends GuStack {
...
-    //   fileName: "recipe-backend-updater.zip",
-    //   timeout: Duration.seconds(30)
-    // })
+    // const crierStreamParam = new GuParameter(this, "crierStream", {
+    //   default: `/${this.stage}/${this.stack}/crier/index-stream`,
+    //   fromSSM: true,
+    //   description: "SSM path to the name of the Crier stream we are attaching to"
+    // });
+
+    new GuKinesisLambdaExperimental(this, "updaterLambda", {
```
Comment (jonathonherbert) [inline cdk/lib/recipes-backend.ts:54]:
```suggestion
    new GuKinesisLambdaExperimental(this, "responderLambda", {
```

... or similar, to match elsewhere? Looks like this is what names the lambda in the AWS dash, which makes it tougher to find

**Review by jonathonherbert — APPROVED** [review]
LGTM, can see ELK log messages in response to CODE edits 👍  one comment on naming might be worth addressing

File: lambda/recipes-responder/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import {EventType} from "@guardian/content-api-models/crier/event/v1/eventType";
+import {ItemType} from "@guardian/content-api-models/crier/event/v1/itemType";
+import type {KinesisStreamHandler, KinesisStreamRecord} from "aws-lambda";
+import {deserializeEvent} from "./deserialize";
+import {handleDeletedContent, handleTakedown} from "./takedown_processor";
+import {handleContentUpdate, handleContentUpdateRetrievable} from "./update_processor";
+
+const filterProductionMonitoring:boolean = process.env["FILTER_PRODUCTION_MONITORING"] ? process.env["FILTER_PRODUCTION_MONITORING"].toLowerCase() =="yes" : false;
```
[the remark this answers is not in the document]
Comment (fredex42) [reply lambda/recipes-responder/src/main.ts:8]: This code is lifted from pubflow-two, I think that the original pubflow uses `yes`/`no` so i copied that approach into there.  No reason we can't use `true` and `false`, but TBH I am thinking to drop this entirely and always filter prodmon stuff here

File: lambda/recipes-responder/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        console.error("ERROR Unknown event type ", evt.eventType);
+    }
+  } catch (err) {
+    console.error(`ERROR Could not process data from Kinesis: ${(err as Error).toString()}`);
+    return null;
+  }
+}
+
+export const handler:KinesisStreamHandler = async (event) => {
+  await Promise.all(event.Records.map((r)=>processRecord(r)))
```
[the remark this answers is not in the document]
Comment (fredex42) [reply lambda/recipes-responder/src/main.ts:48]: That's fair enough, I'd usually do it that way TBH

**Pull request merged**